src/components/views/dialogs/InviteDialog.js

```diff
@@ -351,9 +351,19 @@ export default class InviteDialog extends React.PureComponent {
                 continue;
             }
 
-            const lastEventTs = room.timeline && room.timeline.length
-                ? room.timeline[room.timeline.length - 1].getTs()
-                : 0;
+            // Find the last timestamp for a message event
+            const searchTypes = ["m.room.message", "m.room.encrypted", "m.sticker"];
+            const maxSearchEvents = 20; // to prevent traversing history
+            let lastEventTs = 0;
+            if (room.timeline && room.timeline.length) {
+                for (let i = room.timeline.length - 1; i >= 0; i--) {
+                    const ev = room.timeline[i];
+                    if (searchTypes.includes(ev.getType())) {
+                        lastEventTs = ev.getTs();
```

Comment:
shouldn't we have a break here so it doesn't pick the oldest (within 20) event rather than the latest?

Reply:
oops, yes